vulcanai2/models/basenetwork.py

```diff
@@ -246,13 +250,16 @@ def get_size(self, summary_dict, output):
             summary_dict['output_shape'] = list(output.size())
         return summary_dict
 
-    def get_output_shapes(self, input_size=None):
+    def get_output_shapes(self, network=None, input_size=None):
         """
         Returns the summary of shapes of all layers in the network
         :return: OrderedDict of shape of each layer in the network
         """
         if not input_size:
-            input_size = self._dimensions
+            input_size = self.in_dim
+        # input_size must be a list
+        if isinstance(input_size, (tuple, int)):
```

Comment:
why create a list of a tuple?

Reply:
I think it is more protected  like that, how do you suggest to handle a networks in_dim containing image data, text data and tabular data?
`in_dim=[(conv_in_dim), dense_in_dim] = [(1,28,28), 784]`? Not sure what could be a better alternative

Reply:
you're right. let's just make sure that we have a list of tuples, even for dense_in_dim. e.g. `[(1,28,28), (784)]` so when we interface with it, the type is consistent of the inputs

Reply:
I tried to have the same for dense with `(784)` it reads as a int, it should of been like `(784,)` with a comma, but this implementation is complicating the inputs to the model

Reply:
can't you cast it to list with `list(784)`?

Reply:
then we need to update those areas because we can't have an in_dim=[(1,28,28), 256, (3, 200, 200, 200)] bc each time we go through in_dim we need to check the types instead of just assuming they are all the same. does that make sense?

Reply:
Also, let's say we have an input structure like [conv, dense, dense, conv], so having in_dim like `[(1,28,28), 784, 500, (6, 25, 25)]` will make more sense than, `[(1,28,28), [784, 500], (6, 25, 25)]`, as we use only one for loop to point to the models in self.forward which suits the first case. Does that makes sense?

Reply:
no it should be `[(1,28,28), [784], [500], (6, 25, 25)]`. each in their own list

Reply:
I'm not sure how I can revert back `(784), (500),`to `tuple` form when it gets read as `int`. Doing like  `[784], [500]` will eliminate the purpose of type checking for list of inputs (atleast until if we decide to move to dict type inputs). And as long it is `[(1,28,28), (784), (500), (6, 25, 25)]` or `[(1,28,28), 784, 500, (6, 25, 25)]` `(tuple/int)` it is going to considered as a separate `in_dim`, that's what the type checking is doing there